sources/packages/backend/apps/workers/src/services/supporting-user/supporting-user.service.ts

```diff
@@ -74,4 +75,69 @@ export class SupportingUserService extends RecordDataModelService<SupportingUser
     });
     return entityManager.getRepository(SupportingUser).save(newSupportingUsers);
   }
+
+  /**
+   * Create a new identifiable supporting user if one does not already exist.
+   * @param supportingUser data to create the supporting user.
+   * @param entityManager entity manager to execute in transaction.
+   * @returns the ID of the created or existing supporting user.
+   */
+  async createIdentifiableSupportingUser(
+    supportingUser: IdentifiableSupportingUser,
+    entityManager: EntityManager,
+  ): Promise<number> {
+    const newSupportingUser = new SupportingUser();
+    newSupportingUser.application = {
+      id: supportingUser.applicationId,
+    } as Application;
+    newSupportingUser.supportingUserType = supportingUser.supportingUserType;
+    newSupportingUser.fullName = supportingUser.fullName;
+    newSupportingUser.isAbleToReport = supportingUser.isAbleToReport;
+    newSupportingUser.creator = this.systemUserService.systemUser;
+    const supportingUserRepo = entityManager.getRepository(SupportingUser);
+    const insertResult = await supportingUserRepo
+      .createQueryBuilder()
+      .insert()
+      .into(SupportingUser)
+      .values(newSupportingUser)
+      .orIgnore(
+        "ON CONSTRAINT supporting_users_application_id_full_name DO NOTHING",
+      )
+      .execute();
+    // If the user was created, return its ID.
+    const [identifier] = insertResult.identifiers;
+    if (identifier) {
+      return +identifier.id;
+    }
+    // If the user was not created, it means that it already exists.
+    // Return the ID of the existing user.
+    const existingSupportingUser = await supportingUserRepo.findOne({
+      select: { id: true },
+      where: {
+        application: { id: supportingUser.applicationId },
+        fullName: supportingUser.fullName,
+      },
+    });
+    if (!existingSupportingUser) {
+      // If the user was not created and not found, throw an unexpected error.
+      throw new Error(
+        `Unable to find the supporting user for applicationId: ${supportingUser.applicationId}.`,
+      );
+    }
```

Comment:
Is this validation required? or what are the chances of supporting user not existing at this state?

Reply:
Removed.